App/In-App Actions/URLMenuPresenter.swift

```diff
@@ -292,6 +294,72 @@ private func chromifyURL(_ url: URL) -> URL {
     return components.url!
 }
 
+func downloadVideo(with url: URL, completion: @escaping (URL?) -> ()) {
+    URLSession.shared.downloadTask(with: url) { url, response, error in
+        guard let tempUrl = url, error == nil else {
+            return completion(nil)
+        }
+  
+        let fileManager = FileManager.default
+        let documentsUrl = fileManager.urls(for: .documentDirectory, in: .userDomainMask).first!
+        let videoFileUrl = documentsUrl.appendingPathComponent(response?.suggestedFilename ?? "temp.mp4")
+        
+        if fileManager.fileExists(atPath: videoFileUrl.path) {
+            try? fileManager.removeItem(at: videoFileUrl)
+        }
+        do {
+            try fileManager.moveItem(at: tempUrl, to: videoFileUrl)
+            Log.d("url: \(videoFileUrl)")
+            completion(videoFileUrl)
+        } catch {
+            completion(nil)
+        }
+    }.resume()
+}
+
+func saveToPhotos(_ url: URL, overlay: MRProgressOverlayView?, completion: @escaping (ErrorWorkaround?) -> ()) {
+    PHPhotoLibrary.requestAuthorization { status in
+        guard status == .authorized else {
+            return DispatchQueue.main.async {
+                overlay?.dismiss(true)
+                completion(.accessDenied)
+            }
+        }
+        PHPhotoLibrary.shared().performChanges({
```

Comment:
If we change this to `performChangesAndWait()`, and if we don't bother moving to a temp file, then we don't need to clean up the temp file ourselves. Makes our lives easier.

Reply:
As above, on second thought this is probably best.